File: Task_3/task_3.rb
# Функция для вычисления суммы и произведения цифр числа
def calculate_sum_and_product(n)
    sum = 0
    product = 1
    number_string = n.to_s # Преобразуем число в строку

    number_string.each_char do |digit_char|
        digit = digit_char.to_i # Преобразуем символ цифры в число
        sum += digit
        product *= digit
    end
    return sum, product
end

# Ввод чисел из стандартного ввода
print "Введите числа через пробел: "
input = gets.chomp
number_strings = input.split(" ") # Разбиваем строку на массив чисел
numbers = number_strings.map(&:to_i).filter { |n| n.is_a? Integer } # Преобразуем строки в числа

# Обрабатываем каждое введенное число
numbers.each do |n|
    sum, product = calculate_sum_and_product(n) # Получаем сумму и произведение
    difference = (sum - product).abs # Вычисляем модуль разности

    # Формируем вывод разложения
    number_str = n.to_s
    sum_expression = number_str.chars.map(&:to_i).join(' + ')
    product_expression = number_str.chars.map(&:to_i).join(' * ')

    puts "#{sum} #{product} #{difference} (#{sum_expression} = #{sum}; #{product_expression} = #{product}; |#{sum} - #{product}| = #{difference})"
end

File: Task_3/task_3.swift
import Foundation

// Функция для вычисления суммы и произведения цифр числа
func calculateSumAndProduct(n: Int) -> (sum: Int, product: Int) {
    var sum = 0
    var product = 1
    let numberString = String(n) // Преобразуем число в строку

    for digitChar in numberString {
        if let digit = digitChar.wholeNumberValue { // Преобразуем символ цифры в число
            sum += digit
            product *= digit
        }
    }
    return (sum, product)
}

// Функция для чтения ввода из консоли
func readInput() -> String? {
    let keyboard = FileHandle.standardInput
    let inputData = keyboard.availableData
    let inputString = String(data: inputData, encoding: .utf8)?.trimmingCharacters(in: .whitespacesAndNewlines)
    return inputString
}

// Основная функция
func main() {
    print("Введите числа через пробел:")
    if let input = readInput() {
        let numberStrings = input.components(separatedBy: " ") // Разбиваем строку на массив подстрок
        let numbers = numberStrings.compactMap { Int($0) } // Преобразуем строки в массив чисел

        // Обрабатываем каждое введенное число
        for n in numbers {
            let (sum, product) = calculateSumAndProduct(n) // Получаем сумму и произведение
            let difference = abs(sum - product) // Вычисляем модуль разности

            // Формируем вывод разложения
            let numberStr = String(n)
            let sumExpression = numberStr.compactMap { $0.wholeNumberValue }.map { String($0) }.joined(separator: " + ")
            let productExpression = numberStr.compactMap { $0.wholeNumberValue }.map { String($0) }.joined(separator: " * ")

            print("\(sum) \(product) \(difference) (\(sumExpression) = \(sum); \(productExpression) = \(product); |\(sum) - \(product)| = \(difference))")
        }
    }
}

main()
